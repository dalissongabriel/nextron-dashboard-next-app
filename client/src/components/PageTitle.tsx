import { Divider, Typography } from "@mui/material";
import { PropsWithChildren } from "react";

interface Props {
  bottomDivider?: boolean;
}

export default function PageTitle({
  bottomDivider = false,
  children,
}: PropsWithChildren<Props>) {
  return (
    <>
      <Typography variant="h6" color="primary">
        {children}
      </Typography>
      {bottomDivider && (
        <Divider
          data-testid="page-title-divider-id"
          light
          sx={{ mt: 0.5, mb: 3 }}
        />
      )}
    </>
  );
}
