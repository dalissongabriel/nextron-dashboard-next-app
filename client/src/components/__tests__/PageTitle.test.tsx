import PageTitle from "@components/PageTitle";
import { render, screen } from "@testing-library/react";

describe("PageTitle", () => {
  it("should render title component correctly", () => {
    const { asFragment } = render(<PageTitle>Lorem Ipsum</PageTitle>);

    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByText("Lorem Ipsum")).toBeInTheDocument();
  });

  it("should display divier with sent bottomDivider prop", () => {
    render(<PageTitle bottomDivider>Dolorem amet</PageTitle>);

    expect(screen.getByTestId("page-title-divider-id")).toBeInTheDocument();
  });
});
