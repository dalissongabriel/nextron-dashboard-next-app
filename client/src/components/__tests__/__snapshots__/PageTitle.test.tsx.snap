// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTitle should render title component correctly 1`] = `
<DocumentFragment>
  <h6
    class="MuiTypography-root MuiTypography-h6 css-oblp1g-MuiTypography-root"
  >
    Lorem Ipsum
  </h6>
</DocumentFragment>
`;
